testsuite/integration/manualmode/src/test/java/org/wildfly/test/manual/elytron/seccontext/ElytronClientDefaultSSLContextProviderTestCase.java

```diff
@@ -0,0 +1,95 @@
+package org.wildfly.test.manual.elytron.seccontext;
+
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.wildfly.security.auth.client.ElytronClientDefaultSSLContextProvider;
+import org.wildfly.security.x500.cert.BasicConstraintsExtension;
+import org.wildfly.security.x500.cert.X509CertificateBuilder;
+
+import javax.net.ssl.SSLContext;
+import javax.security.auth.x500.X500Principal;
+import javax.ws.rs.client.ClientBuilder;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.Security;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+public class ElytronClientDefaultSSLContextProviderTestCase {
+
+    private static final String CONFIG_FILE = "file:./src/test/resources/wildfly-config-default-ssl-context.xml";
+    private static final String CLIENT_KEYSTORE = "client.keystore";
+    private static final char[] PASSWORD = "password".toCharArray();
+    private static final String ALIAS = "alias";
+    private static File KEYSTORES_DIR = new File("./target/keystores");
+
+    @BeforeClass
+    public static void generateKeystoreAndConfigureProvider() throws CertificateException, NoSuchAlgorithmException, KeyStoreException, IOException {
+        new File(KEYSTORES_DIR, CLIENT_KEYSTORE).delete();
+        KEYSTORES_DIR.delete();
+        generateClientKeystore();
+        Security.insertProviderAt(new ElytronClientDefaultSSLContextProvider(CONFIG_FILE), 1);
+    }
+
+    @Test
+    public void testDefaultSSLContextIsNotReturnedIfNotConfigured() {
```

Comment:
I am not sure that the test name is actually reflecting what is checked, here. Or am I missing something?

Also, could you please add Javadoc?